### PR TITLE
Created base clients for ChangeFeed

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/module-info.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
     exports com.azure.storage.blob.implementation to
         com.azure.storage.blob.cryptography,
         com.azure.storage.blob.batch,
-        com.azure.storage.file.datalake;    // FIXME this should not be a long-term solution
+        com.azure.storage.file.datalake, azure.storage.blob.changefeed;    // FIXME this should not be a long-term solution
```
[the remark this answers is not in the document]
Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/main/java/module-info.java]: done

File: sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedAsyncClient.java [hunk 2]
```diff
@@ -4,10 +4,38 @@
...
 import com.azure.core.annotation.ServiceClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.BlobContainerClientBuilder;
+import com.azure.storage.blob.BlobServiceVersion;
 
 /**
- * TEMP
+ * This class provides a client that contains all operations that apply to Azure Storage Blob changefeed.
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedAsyncClient.java]: I'd like to have more informative docs. Not sure if you were planning on adding that later.

File: sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedClientBuilder.java [hunk 3]
```diff
@@ -4,10 +4,72 @@
...
+import com.azure.core.http.HttpPipeline;
+import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceVersion;
 
 /**
- * TEMP
+ * This class provides a fluent builder API to help aid the configuration and instantiation of
+ * {@link BlobChangefeedClient BlobChangefeedClients} and {@link BlobChangefeedAsyncClient BlobChangefeedAsyncClients}
+ * when {@link #buildClient() buildClient} and {@link #buildAsyncClient() buildAsyncClient} as called respectively.
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedClientBuilder.java]: "as called respectively"

File: sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedClientBuilder.java [hunk 4]
```diff
@@ -4,10 +4,72 @@
...
 @ServiceClientBuilder(serviceClients = {BlobChangefeedClient.class, BlobChangefeedAsyncClient.class})
-public class BlobChangefeedClientBuilder {
+public final class BlobChangefeedClientBuilder {
+    private final String accountUrl;
+    private final HttpPipeline pipeline;
+    private final BlobServiceVersion version;
+
+    /**
+     * Constructs the {@link BlobChangefeedClientBuilder} using the {@link BlobServiceClient#getAccountUrl() account
+     * URL} and {@link BlobServiceClient#getHttpPipeline() HttpPipeline} properties of the passed {@link
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedClientBuilder.java]: I think we can just say using the URL (or endpoint) and pipeline of the passed client. I don't think we need to specify that we use the getters.

File: sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedClientBuilder.java [hunk 5]
```diff
@@ -4,10 +4,72 @@
...
+     * @param client {@link BlobServiceClient} whose properties are used to configure the builder.
+     */
+    public BlobChangefeedClientBuilder(BlobServiceAsyncClient client) {
+        this.accountUrl = client.getAccountUrl();
+        this.pipeline = client.getHttpPipeline();
+        this.version = client.getServiceVersion();
+    }
+
+    /**
+     * Creates a {@link BlobChangefeedClient} based on options set in the builder.
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedClientBuilder.java]: Are we planning on adding options? Or should we change the doc?

File: sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedAsyncClient.java (hunk 2 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedAsyncClient.java]: yep, I will add more information as I understand more about ChangeFeed and add more related classes.

File: sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedClientBuilder.java (hunk 5 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedClientBuilder.java]: Let me change the doc for now and add it if we do end up adding options

**Pull request merged**